contract/services.py

```diff
@@ -415,6 +415,39 @@ def delete(self, contract):
         except Exception as exc:
             return _output_exception(model_name="Contract", method="delete", exception=exc)
 
+    @check_authentication
+    def terminate_contract(self, contract):
+        try:
+            # TODO - add this service to the tasks.py in apscheduler once a day
+            #  to check if contract might be terminated
+            contract_to_terminate = ContractModel.objects.filter(id=contract["id"]).first()
+            from core import datetime, datetimedelta
+            now = datetime.datetime.now()
+            if contract_to_terminate.date_valid_to < now:
+                # we can marked that contract as a terminated
+                contract_to_terminate.state = ContractModel.STATE_TERMINATED
```

Comment:
my misstake here, it is for STATE_EXECUTED if state was 7

Reply:
Ok, I also can add to verify if STATE_EXECUTED is etc

Reply:
Obviously I will implement as below with simply filtering according to your proposition: ContractModel.objects.filter(date_valid_to__lt=datetime.date.now()).filter(state=7)